internal/storage/v2/v1adapter/translator.go

```diff
@@ -63,6 +63,134 @@ func V1TracesFromSeq2(otelSeq iter.Seq2[[]ptrace.Traces, error]) ([]*model.Trace
 	return jaegerTraces, nil
 }
 
+// V1TracesFromSeq2WithLimit converts an interator of ptrace.Traces chunks into v1 traces.
+// If maxTraceSize > 0, traces exceeding that number of spans will be truncated while consuming the sequence.
+func V1TracesFromSeq2WithLimit(otelSeq iter.Seq2[[]ptrace.Traces, error], maxTraceSize int) ([]*model.Trace, error) {
```

Comment:
if the truncation logic is in applyTraceSizeLimit, then this V1TracesFromSeq2WithLimit method is completely redundant, you can achieve the same result by combining applyTraceSizeLimit with V1TracesFromSeq2